ingestor/adx/tasks.go

```diff
@@ -163,10 +164,30 @@ func (t *SyncFunctionsTask) Run(ctx context.Context) error {
 			}
 		}
 
+		t.cache[cacheKey] = function
 		logger.Infof("Successfully created function %s.%s", function.Spec.Database, function.Name)
 		updateKQLFunctionStatus(ctx, t.store, function, v1.Success, nil)
 	}
 
+	for cacheKey, fn := range t.cache {
+		if !slices.ContainsFunc(functions, func(i *v1.Function) bool {
+			return fn.Name == i.Name && fn.Spec.Database == i.Spec.Database
+		}) {
+			logger.Warnf("Function %s.%s is no longer in the store, deleting", fn.Spec.Database, fn.Name)
+			stmt := kql.New(".drop function ").AddUnsafe(fn.Name)
+			if _, err := t.kustoCli.Mgmt(ctx, stmt); err != nil {
+				if !errors.Retry(err) {
+					logger.Errorf("Failed to delete function %s.%s: %v", fn.Spec.Database, fn.Name, err)
+					delete(t.cache, cacheKey)
+				} else {
+					logger.Warnf("Transient failure to delete function %s.%s: %v", fn.Spec.Database, fn.Name, err)
+					continue
+				}
+			}
+			delete(t.cache, cacheKey)
+		}
+	}
+
```

Comment:
Supports deleting a Function from Kusto when its corresponding CRD is deleted from k8s.